### PR TITLE
[EOS-16538] Created S3_IAM_REST_API_&_CLI.md

### PR DESCRIPTION
**Pull request by Saumya-Sunder**
Created S3_IAM_REST_API_&_CLI.md based on the Jira https://jts.seagate.com/browse/EOS-16538

Signed-off-by: Saumya Sunder <saumya.sunder@seagate.com>

-----
[View rendered docs/S3_IAM_REST_API_&_CLI.md](https://github.com/Saumya-Sunder/cortx-s3server-1/blob/patch-8/docs/S3_IAM_REST_API_&_CLI.md)

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Review by johnbent — CHANGES_REQUESTED** [review]
Please add a link from an existing document to this newly added file.

FYI @nileshgovande , @shalakadharap , @amitwac2608 , this is our general rule for working on documents.  If we do not enforce this rule, we will surely forget sometimes to subsequently add the link and we will end up with orphaned files.  Orphaned files are of no value because very few people read the docs by looking at the directory view.  Instead, they read docs and follow links.

Thanks!

**Comment by nileshgovande** [comment]
Looks ggod on a first look - formatting in particular.
Tagging @bkirunge7 to review the document

File: docs/S3_IAM_REST_API_&_CLI.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
+# Supported S3 IAM REST APIs and CLI Operations
```
Comment (amitwac2608) [inline docs/S3_IAM_REST_API_&_CLI.md:1]: What about this:

Supported S3 IAM REST APIs and CLI operations in CORTX

File: docs/S3_IAM_REST_API_&_CLI.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+The S3 IAM REST APIs are a collection of resources and methods which are integrated with AWS Identity and Access Management (IAM) service. The IAM service lets an administrator to securely control access to the S3 resources. As an IAM administrator, you can control the authentication and authorizations for an ID to use API Gateway resources. We've listed the S3 IAM REST APIs and CLI operations that are supported by the CORTX-S3 Server Submodule.
```
Comment (amitwac2608) [inline docs/S3_IAM_REST_API_&_CLI.md:3]: Should we use AWS term ? @bkirunge7

File: docs/S3_IAM_REST_API_&_CLI.md [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+
+The S3 IAM REST APIs are a collection of resources and methods which are integrated with AWS Identity and Access Management (IAM) service. The IAM service lets an administrator to securely control access to the S3 resources. As an IAM administrator, you can control the authentication and authorizations for an ID to use API Gateway resources. We've listed the S3 IAM REST APIs and CLI operations that are supported by the CORTX-S3 Server Submodule. 
+
+<details>
+<summary>CreateAccount</summary>
+<p>
+
+The CreateAccount request lets you create an S3 IAM account. 
+
+| Request | Request body attributes  | Request Parameters    |
```
Comment (amitwac2608) [inline docs/S3_IAM_REST_API_&_CLI.md:11]: some APIs have response section as well. I guess we add response section to all APIs

Comment (Saumya-Sunder) [reply docs/S3_IAM_REST_API_&_CLI.md:11]: I have mentioned the response below the tables, try expanding a few and you will see them @amitwac2608

Comment (amitwac2608) [reply docs/S3_IAM_REST_API_&_CLI.md:11]: yes Saumya, this is what I mentioned  - few APIs have response section, not all. 

So I added comment that "should we add response section for all APIs ? "

File: docs/S3_IAM_REST_API_&_CLI.md (hunk 1 above)
Comment (Saumya-Sunder) [reply docs/S3_IAM_REST_API_&_CLI.md]: Done

File: docs/S3_IAM_REST_API_&_CLI.md (hunk 3 above)
Comment (Saumya-Sunder) [reply docs/S3_IAM_REST_API_&_CLI.md:11]: I think we definitely should. Who can help me with these?

Comment (Saumya-Sunder) [reply docs/S3_IAM_REST_API_&_CLI.md:11]: file://shgpiwfsw001.ad.seagate.com/Data/753287/Downloads/S3%20IAM%20REST%20API.pdf

**Comment by Saumya-Sunder** [comment]
Hey @johnbent just checked this file out and looks like it is connected via the S3 README > CortxS3_Architecture.md doc "Our S3 Compatability Matrix can be found [here."](https://github.com/Saumya-Sunder/cortx-s3server-1/blob/patch-8/docs/s3-supported-api.md).

Please let me know if you would want me to link it from any other place. 

@nileshgovande, @amitwac2608, @bkirunge7 please let me know your inputs so that I could incorporate the changes to the document.

> Please add a link from an existing document to this newly added file.
> 
> FYI @nileshgovande , @shalakadharap , @amitwac2608 , this is our general rule for working on documents. If we do not enforce this rule, we will surely forget sometimes to subsequently add the link and we will end up with orphaned files. Orphaned files are of no value because very few people read the docs by looking at the directory view. Instead, they read docs and follow links.
> 
> Thanks!

**Pull request closed without merging**